projects/js-packages/charts/src/charts/sparkline/sparkline.tsx

```diff
@@ -0,0 +1,177 @@
+ * @return SeriesData array suitable for LineChart
+ */
+const transformToSeriesData = (
+	data: number[],
+	color?: string,
+	strokeWidth?: number
+): SeriesData[] => {
+	// Use a fixed base date and increment by index to create linear spacing
+	const baseDate = new Date( 2000, 0, 1 );
+
+	return [
+		{
+			label: 'sparkline',
+			data: data.map( ( value, index ) => ( {
+				date: new Date( baseDate.getTime() + index * 86400000 ), // Add days
+				value,
+			} ) ),
+			options: {
+				stroke: color,
+				seriesLineStyle: strokeWidth ? { strokeWidth } : undefined,
+			},
+		},
+	];
+};
+
+const SparklineComponent = forwardRef< HTMLDivElement, SparklineProps >(
+	(
+		{
+			data,
+			width = DEFAULT_WIDTH,
+			height = DEFAULT_HEIGHT,
+			color,
+			strokeWidth = DEFAULT_STROKE_WIDTH,
+			withGradientFill = true,
+			gradient,
+			className,
+			chartId,
+			margin = DEFAULT_MARGIN,
+		},
+		ref
+	) => {
+		// Transform number[] to SeriesData[] for LineChart
+		const seriesData = useMemo( () => {
+			if ( ! data || data.length === 0 ) {
+				return [];
+			}
+			return transformToSeriesData( data, color, strokeWidth );
+		}, [ data, color, strokeWidth ] );
+
+		// Merge margins
+		const finalMargin = useMemo( () => {
+			return {
+				...DEFAULT_MARGIN,
+				...margin,
+			};
+		}, [ margin ] );
+
+		// Build gradient options for the series if custom gradient is provided
+		// Note: This must be called before any early returns to follow React hooks rules
+		const seriesWithGradient = useMemo( () => {
+			if ( ! gradient || seriesData.length === 0 ) {
+				return seriesData;
+			}
+
+			return seriesData.map( series => ( {
+				...series,
+				options: {
+					...series.options,
+					gradient: {
+						from: gradient.from || color || '#000000',
+						to: gradient.to || '#ffffff',
+						fromOpacity: gradient.fromOpacity ?? 0.5,
+						toOpacity: gradient.toOpacity ?? 0.0,
+					},
+				},
+			} ) );
+		}, [ seriesData, gradient, color ] );
+
+		// Handle empty data
+		if ( ! data || data.length === 0 ) {
+			return (
+				<div
+					ref={ ref }
+					className={ clsx( styles.sparkline, styles[ 'sparkline--empty' ], className ) }
+					style={ { width, height } }
+					data-testid="sparkline-empty"
+				/>
+			);
+		}
+
+		// Handle single data point - render a simple dot
+		if ( data.length === 1 ) {
+			const cx = width / 2;
+			const cy = height / 2;
+			const resolvedColor = color || '#000000';
+
+			return (
+				<div
+					ref={ ref }
+					className={ clsx( styles.sparkline, styles[ 'sparkline--single-point' ], className ) }
+					style={ { width, height } }
+					data-testid="sparkline-single-point"
+				>
+					<svg width={ width } height={ height } aria-hidden="true">
+						<circle cx={ cx } cy={ cy } r={ strokeWidth * 1.5 } fill={ resolvedColor } />
+					</svg>
+				</div>
+			);
+		}
+
+		return (
+			<div ref={ ref } className={ clsx( styles.sparkline, className ) } data-testid="sparkline">
```

Comment:
We normally add the base className here too
```suggestion
			<div ref={ ref } className={ clsx( 'sparkline', styles.sparkline, className ) } data-testid="sparkline">
```